### DISCUSSION TITLE
ESP32 in bootloop

### DISCUSSION THREAD

**Discussion by ThreeSixty123** [jomjol/AI-on-the-edge-device#878] [Q&A] [answered]
Hi,

I've made everything like in description.

Erasing, Flashing, rebooting etc.

With Flashtool V. 3.9.2 and also with python / powershell ... everything looks normal but after disconnecting the powercable and connecting it again i do not see the LED on, Wifi is not having a new device and i hear  (round about) 10 beeps / ticks and a pause of a  second than starting all over again.....

I am using dev board to connect  and chip is ESP32S but trying to conncet with ESP32 download tool i can only use esp32

**Comment by ghost** [comment]
Could you add a serial connection and paste the logging here?

**Comment by ThreeSixty123** [comment, marked as answer]
Hi Frank,

I Just Changes the SD Card ... 3rd one with 16 GB worked. Somehow the
others two were not compatible....

Thanks for prompt Feedback.

Mfg
Richard


Frank Haverland ***@***.***> schrieb am Mo., 25. Juli 2022,
15:14:

**Reply by ghost** [reply]
The ESP32 supports only SD-cards up to 4gb. After a while I had the same issues and use now only 4gb cards.

**Reply by caco3** [reply]
Odly, I use 16 GB cards and it works fine as of now.

**Comment by mundi86** [comment]
Hi,

i need help :(
With Flashtool V. 3.9.2 i flashed the firmware (diverent version) and on the SD Card i copied everthing (4gb/8gb/16gb/32gb --- FAT32)
but nothing works... 
i order this esp32-cam -> https://www.amazon.de/gp/product/B0B56BDBMC/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&psc=1

what is wrong with this:  
[0;32mI (844) psram: This chip is ESP32-D0WD[0m
[0;31mE (845) psram: PSRAM ID read error: 0xffffffff[0m
[0;31mE (845) cpu_start: Failed to init external RAM![0m

----------------------------------------------------------
Seriel-Monitor in ArduinoIDE shows me:
----------------------------------------------------------
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7664
load:0x40078000,len:15360
load:0x40080400,len:3744
entry 0x4008067c
[0;32mI (27) boot: ESP-IDF 4.3.2 2nd stage bootloader[0m
[0;32mI (27) boot: compile time 18:16:43[0m
[0;32mI (27) boot: chip revision: 3[0m
[0;32mI (30) boot_comm: chip revision: 3, min. bootloader chip revision: 0[0m
[0;32mI (37) boot.esp32: SPI Speed      : 40MHz[0m
[0;32mI (42) boot.esp32: SPI Mode       : DIO[0m
[0;32mI (46) boot.esp32: SPI Flash Size : 4MB[0m
[0;32mI (51) boot: Enabling RNG early entropy source...[0m
[0;32mI (56) boot: Partition Table:[0m
[0;32mI (60) boot: ## Label            Usage          Type ST Offset   Length[0m
[0;32mI (67) boot:  0 nvs              WiFi data        01 02 00009000 00004000[0m
[0;32mI (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000[0m
[0;32mI (82) boot:  2 phy_init         RF data          01 01 0000f000 00001000[0m
[0;32mI (89) boot:  3 ota_0            OTA app          00 10 00010000 001db000[0m
[0;32mI (97) boot:  4 ota_1            OTA app          00 11 001f0000 001db000[0m
[0;32mI (104) boot: End of partition table[0m
[0;32mI (109) boot_comm: chip revision: 3, min. application chip revision: 0[0m
[0;32mI (116) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=4b4b8h (308408) map[0m
[0;32mI (242) esp_image: segment 1: paddr=0005b4e0 vaddr=3ffb0000 size=04a44h ( 19012) load[0m
[0;32mI (250) esp_image: segment 2: paddr=0005ff2c vaddr=40080000 size=000ech (   236) load[0m
[0;32mI (251) esp_image: segment 3: paddr=00060020 vaddr=400d0020 size=146c74h (1338484) map[0m
[0;32mI (769) esp_image: segment 4: paddr=001a6c9c vaddr=400800ec size=1bb90h (113552) load[0m
[0;32mI (818) esp_image: segment 5: paddr=001c2834 vaddr=50000000 size=00010h (    16) load[0m
[0;32mI (832) boot: Loaded app from partition at offset 0x10000[0m
[0;32mI (832) boot: Disabling RNG early entropy source...[0m
[0;32mI (844) psram: This chip is ESP32-D0WD[0m
[0;31mE (845) psram: PSRAM ID read error: 0xffffffff[0m
[0;31mE (845) cpu_start: Failed to init external RAM![0m
Re-enable cpu cache.

abort() was called at PC 0x40082d6b on core 0

Backtrace:0x40082c01:0x3ffe3ba0 0x4008f635:0x3ffe3bc0 0x40096419:0x3ffe3be0 0x40082d6b:0x3ffe3c50 0x4007956d:0x3ffe3c80 |<-CORRUPTED


ELF file SHA256: 400543e46edf0eca

Rebooting...

**Reply by jomjol** [reply]
Looks like a broken PSRAM:
```
�[0;31mE (845) psram: PSRAM ID read error: 0xffffffff�[0m
�[0;31mE (845) cpu_start: Failed to init external RAM!�[0m
```

This has been reported from time to time and is a HW-failure.

**Reply by mundi86** [reply]
Hi Jomjol!

thx 4 the quick anwser, I've tried another (because I thought the first one might be damaged, and you think this to), but I am seeing the same behavior there. Is there anything I can do to troubleshoot the problem and find the root cause?

or should i buy another one from a different seller to teste it?

**Reply by wimg** [reply]
Seeing the same problem here with one ordered from Amazon ( https://www.amazon.de/gp/product/B08G132D3J/ref=ppx_yo_dt_b_asin_title_o05_s00?ie=UTF8&psc=1 )
Is this a problem with the device or could it be a recent change in the firmware that broke this ?

Output is :

```
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7664
load:0x40078000,len:15360
load:0x40080400,len:3744
entry 0x4008067c
I (27) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (27) boot: compile time 20:02:47
I (27) boot: chip revision: 3
I (30) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (37) boot.esp32: SPI Speed      : 40MHz
I (42) boot.esp32: SPI Mode       : DIO
I (46) boot.esp32: SPI Flash Size : 4MB
I (51) boot: Enabling RNG early entropy source...
I (56) boot: Partition Table:
I (60) boot: ## Label            Usage          Type ST Offset   Length
I (67) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (82) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (89) boot:  3 ota_0            OTA app          00 10 00010000 001db000
I (97) boot:  4 ota_1            OTA app          00 11 001f0000 001db000
I (104) boot: End of partition table
I (109) boot_comm: chip revision: 3, min. application chip revision: 0
I (116) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=4b88ch (309388) map
I (243) esp_image: segment 1: paddr=0005b8b4 vaddr=3ffb0000 size=04294h ( 17044) load
I (250) esp_image: segment 2: paddr=0005fb50 vaddr=40080000 size=004c8h (  1224) load
I (251) esp_image: segment 3: paddr=00060020 vaddr=400d0020 size=16668ch (1468044) map
I (819) esp_image: segment 4: paddr=001c66b4 vaddr=400804c8 size=1b1d0h (111056) load
I (867) esp_image: segment 5: paddr=001e188c vaddr=50000000 size=00010h (    16) load
I (881) boot: Loaded app from partition at offset 0x10000
I (881) boot: Disabling RNG early entropy source...
I (893) psram: This chip is ESP32-D0WD
E (893) psram: PSRAM ID read error: 0xffffffff
E (893) cpu_start: Failed to init external RAM!

abort() was called at PC 0x400829e1 on core 0


Backtrace:0x40083265:0x3ffe3ba00x4008f07d:0x3ffe3bc0 0x40096211:0x3ffe3be0 0x400829e1:0x3ffe3c50 0x4007956d:0x3ffe3c80  |<-CORRUPTED




ELF file SHA256: cb5bbb12254b3d09

```

**Reply by mundi86** [reply]
hi wimg,

i checked the link and thats the same that i orderd.
in the details i found: "**_Eingebauter 520 KB SRAM, externes 4M PSRAM_**"
I think it should be 8MB PSRAM ?!?!?!?
So i orderd 2 new esp-32 cam with 8MB 
https://www.amazon.de/gp/product/B08P1NMPLL/ref=ppx_yo_dt_b_asin_title_o09_s00?ie=UTF8&psc=1

but after some checks, they also have only 4 MB....
https://photos.app.goo.gl/kbhAer8MCH27P4y6A

i dont know.... nothing is working ....

**Comment by jomjol** [comment]
if your sure about the flashing, than there is nothing else that you can do, but just exchange the hardware.

**Reply by mundi86** [reply]
Hi, thx for the feedback :)
i orderd 2 new esp32-cam.
crossing fingers that they are working fine :)
thx
